tests/Microsoft.Identity.Test.Common/Core/Mocks/MockHttpManagerExtensions.cs

```diff
@@ -468,11 +484,44 @@ private static MockHttpMessageHandler BuildMockHandlerForManagedIdentitySource(M
                     break;
             }
 
+            var manager = new CommonCryptographyManager();
+
+            // If capabilityEnabled, add "xms_cc": "cp1"
+            if (capabilityEnabled)
+            {
+                if (managedIdentitySourceType == ManagedIdentitySource.ServiceFabric)
+                {
+                    expectedQueryParams.Add("xms_cc", "cp1,cp2");
+                }
+            }
+            else
+            {
+                notExpectedQueryParams.Add("xms_cc", "cp1,cp2");
+            }
+
+            if (claimsEnabled)
+            {
+                if (managedIdentitySourceType == ManagedIdentitySource.ServiceFabric)
```

Comment:
I liked how this is handled by a list in Java and JS by creating a list of MI sources that support claims and capabilities. Then you can use this when adding claims and capabilities in other managed identity sources with less code changes.